Add changes to UPGRADE file (#454)
Sylius/RefundPlugin: UPGRADE-2.0.md

Reviewer: I believe there are some other aliases/ids that has been replaced and can be added here:
https://github.com/Sylius/RefundPlugin/pull/454/files#diff-57a2d391e31e0d1c39e9a12a231e9e3cb7e2554b7faa11c80ab43b7ce889c631R139-R144

1. Doctrine migrations have been regenerated, meaning all previous migration files have been removed and their content is now in a single migration file.
   To apply the new migration and get rid of the old entries run migrations as usual:

   ```bash
       bin/console doctrine:migrations:migrate --no-interaction
   ```

1. The directories structure has been updated to the current Symfony recommendations:
   - `@SyliusRefundPlugin/Resources/config` -> `@SyliusRefundPlugin/config`
   - `@SyliusRefundPlugin/Resources/translations` -> `@SyliusRefundPlugin/translations`
   - `@SyliusRefundPlugin/Resources/views` -> `@SyliusRefundPlugin/templates`

   You need to adjust the import of configuration file in your end application:
   ```diff
   imports:
   -   - { resource: "@SyliusRefundPlugin/Resources/config/config.yml" }
   +   - { resource: '@SyliusRefundPlugin/config/config.yaml' }
   ```

   And the routes configuration paths:
   ```diff
   sylius_refund:
   -   resource: "@SyliusRefundPlugin/Resources/config/routing.yml"

>    +   resource: "@SyliusRefundPlugin/config/routing.yml"

Reviewer 2: Maybe let's change the file extension to `yaml`.

   ```

   Adjust the paths to assets and templates if you are using them.

1. No need to overwrite templates:  
   Thanks to the use of Twig Hooks and the refactoring of templates, you no longer need to overwrite templates to use plugin features.

1. The following routes have been removed:  
    - `sylius_refund_order_credit_memos_list`

>     - `sylius_refund_plugin_shop_order_credit_memos_partial`

Reviewer 3: I can still see these routes in the 2.0 branch.

Author: Changes in https://github.com/Sylius/RefundPlugin/pull/458

> 1. Aliases introduced in RefundPlugin 1.6 have now become the primary service IDs in RefundPlugin 2.0.

Reviewer 3: Redundant, as the same note is mentioned below.

1. Doctrine migrations have been regenerated, meaning all previous migration files have been removed and their content is now in a single migration file.
   To apply the new migration and get rid of the old entries run migrations as usual:

   ```bash
       bin/console doctrine:migrations:migrate --no-interaction
   ```

1. The directories structure has been updated to the current Symfony recommendations:
   - `@SyliusRefundPlugin/Resources/config` -> `@SyliusRefundPlugin/config`
   - `@SyliusRefundPlugin/Resources/translations` -> `@SyliusRefundPlugin/translations`
   - `@SyliusRefundPlugin/Resources/views` -> `@SyliusRefundPlugin/templates`

   You need to adjust the import of configuration file in your end application:
   ```diff
   imports:
   -   - { resource: "@SyliusRefundPlugin/Resources/config/config.yml" }
   +   - { resource: '@SyliusRefundPlugin/config/config.yaml' }
   ```

   And the routes configuration paths:
   ```diff
   sylius_refund:
   -   resource: "@SyliusRefundPlugin/Resources/config/routing.yml"
   +   resource: "@SyliusRefundPlugin/config/routing.yml"
   ```

>    And the paths to assets and templates if you are using them.

Reviewer: ```suggestion
   Adjust the paths to assets and templates if you are using them.
```

> 

Reviewer: ```suggestion

```

   - Services required by Symfony to be `public` (e.g., controllers, event listeners) remain public.

1. `_javascript.html.twig` file has been removed, and its code has been moved to `src/Resources/assets/js/refund-button.js`. When upgrading to 2.0, import the `src/Resources/assets/entrypoint.js` file into your application’s main js file.

1. The following constructor signatures have been changed:

    - `Sylius\RefundPlugin\Action\Admin\OrderRefundsListAction`:
        ```diff
        public function __construct(
        -       private readonly SessionInterface | RequestStack $requestStackOrSession,
        +       private RequestStack $requestStack,
        ) {
        }
        ```

    - `Sylius\RefundPlugin\Action\Admin\RefundUnitsAction`:
        ```diff
        public function __construct(
        -       private readonly SessionInterface|RequestStack $requestStackOrSession,
        -       private readonly RequestCommandCreatorInterface|RefundUnitsCommandCreatorInterface $commandCreator,
        +       private RequestStack $requestStack,

>         +       private RequestCommandCreatorInterface $commandCreator

Reviewer: ```suggestion
        +       private RequestCommandCreatorInterface $commandCreator,
```

        ) {
        }
        ```

    - `Sylius\RefundPlugin\Action\Admin\SendCreditMemoAction`:
        ```diff
        public function __construct(
        -       private readonly SessionInterface | RequestStack $requestStackOrSession,
        +       private RequestStack $requestStack,
        ) {
        }
        ```

   - `Sylius\RefundPlugin\CommandHandler\GenerateCreditMemoHandler`:
       ```diff
       public function __construct(
       -       private readonly ObjectManager $creditMemoManager,
       -       private readonly ?CreditMemoFileResolverInterface $creditMemoFileResolver = null,
       +       private EntityManagerInterface $creditMemoManager,
       +       private CreditMemoFileResolverInterface $creditMemoFileResolver,
       ) {
       }
       ```

   - `Sylius\RefundPlugin\CommandHandler\RefundUnitsHandler`:
       ```diff
       public function __construct(
       -    private EntityManagerInterface $entityManager,
       +    private ObjectManager $entityManager,
       ) {
       }

>        ```

Reviewer: Sth has gone wrong here, because I don't see such a constructor for this handler 🤔

Reviewer: What is more, `RefundUnitsHandler` looks like it needs some fixes in 2.0

Author: Fixed

   - `Sylius\RefundPlugin\Converter\LineItem\OrderItemUnitLineItemsConverter`:
       ```diff
       public function __construct(
       -       private readonly ?LineItemFactoryInterface $lineItemFactory = null,
       +       private LineItemFactoryInterface $lineItemFactory,
       ) {
       }
       ```

   - `Sylius\RefundPlugin\Converter\LineItem\ShipmentLineItemsConverter`:
       ```diff
       public function __construct(
       -       private readonly ?LineItemFactoryInterface $lineItemFactory = null,
       +       private LineItemFactoryInterface $lineItemFactory,
       ) {
       }
       ```

   - `Sylius\RefundPlugin\Creator\RefundCreator`:
       ```diff
       public function __construct(
       -    private EntityManagerInterface $entityManager,

>        +    private ObjectManager $entityManager,

Reviewer: 😅 
```suggestion
       -    private ObjectManager $entityManager,
       +    private EntityManagerInterface $entityManager,
```

       -       private readonly ?UnitRefundFilterInterface $unitRefundFilter = null,
       +       private UnitRefundFilterInterface $unitRefundFilter,
       ) {
       }
       ```

    - `Sylius\RefundPlugin\Sender\CreditMemoEmailSender`:
       ```diff
       public function __construct(
       -       private readonly ?CreditMemoPdfFileGeneratorInterface $creditMemoPdfFileGenerator,
       -       private readonly ?FileManagerInterface $fileManager,
       -       private readonly ?CreditMemoFileResolverInterface $creditMemoFileResolver = null,
       -       private readonly ?CreditMemoFilePathResolverInterface $creditMemoFilePathResolver = null,
       +       private CreditMemoFileResolverInterface $creditMemoFileResolver,
       +       private CreditMemoFilePathResolverInterface $creditMemoFilePathResolver,
       ) {
       }
       ```

   - `Sylius\RefundPlugin\Validator\RefundUnitsCommandValidator`:
      ```diff
      public function __construct(
      -       private ?iterable $refundUnitsBelongingToOrderValidators = null,
      +       private iterable $refundUnitsBelongingToOrderValidators,
      ) {
      }
      ```   
      

> 1. The following classes where deprecated in 1.x and have been removed in 2.0:

Reviewer: ```suggestion
1. The following deprecated classes and interfaces have been removed in 2.0:
```
